include/static-pages.php

```diff
@@ -180,10 +180,11 @@ public function translate_page_for_posts( $page_id ) {
 	 */
 	public function oembed_request_post_id( $post_id, $url ) {
 		foreach ( $this->model->get_languages_list() as $lang ) {
-			if ( trailingslashit( $url ) === trailingslashit( $lang->home_url ) ) {
+			if ( is_string( $lang->home_url ) && is_int( $lang->page_on_front ) && ( trailingslashit( $url ) === trailingslashit( $lang->home_url ) ) ) {
 				$post_id = $lang->page_on_front;
 			}
 		}
+
 		return $post_id;
```

Comment:
Another very long `if` imho:
```suggestion
			if ( ! is_string( $lang->home_url ) ) {
				continue;
			}
			
			if ( trailingslashit( $url ) === trailingslashit( $lang->home_url ) ) {
				return (int) $lang->page_on_front;
			}
		}

		return $post_id;
```
One slight change from previous behavior: we return the "page on front" post ID as soon as we have a match, instead of continuing the loop for no reason.

Reply:
Good idea! Thanks!